File: backend/Wms.Domain/Entities/Product.cs
namespace Wms.Domain.Entities;

public class Product
{
    public int Id { get; set; }
    public required string Sku { get; set; }
    public required string Name { get; set; }
    public string? Barcode { get; set; }
    public bool IsActive { get; set; } = true;
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
}

File: frontend/wms-web/src/types/product.ts
export type Product = {
  id: number;
  sku: string;
  name: string;
  barcode: string | null;
  minimumStock: number;
  isActive: boolean;
  createdAtUtc: string;
};

export type CreateProductRequest = {
  sku: string;
  name: string;
  barcode: string | null;
  minimumStock: number;
};

export type UpdateProductRequest = {
  sku: string;
  name: string;
  barcode: string | null;
  minimumStock: number;
  isActive: boolean;
};
